### PR TITLE
tkt-54954: Add enable_smb1 checkbox

### PR DESCRIPTION
**Pull request by anodos325**
tkt: 54954 
- add checkbox to enable_smb1
- remove UI change where we were hiding "unix extensions" based on value of server min protocol sysctl
- if sysctl is set to enable NT1, then check the "enable smb1" box when we generate the smb4.conf file.

### CODE REVIEW

File: gui/services/migrations/0020_add_enable_smb1.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+def move_sysctl_min_protocol(apps, schemaeditor):
+    tunables = apps.get_model('system', 'tunable')
+    smb1_sysctl = tunables.filter(tun_type='SYSCTL',
+                                  tun_var='freenas.services.smb.config.server_min_protocol',
+                                  tun_value='NT1')
+
+    if smb1_sysctl.exists():
+        cifs.cifs_srv_enable_smb1 = True
+        cifs.save
```
Comment (william-gr) [inline gui/services/migrations/0020_add_enable_smb1.py]: .save()

File: src/freenas/usr/local/libexec/nas/generate_smb4_conf.py [hunk 2]
```diff
@@ -952,9 +952,8 @@ def generate_smb4_conf(client, smb4_conf, role):
     if os.path.exists("/usr/local/etc/smbusers"):
         confset1(smb4_conf, "username map = /usr/local/etc/smbusers")
 
-    server_min_protocol = fs().services.smb.config.server_min_protocol
-    if server_min_protocol != 'NONE':
-        confset2(smb4_conf, "server min protocol = %s", server_min_protocol)
+    if not cifs.enable_smb1:
+        confset2(smb4_conf, "server min protocol = SMB2_02")
```
Comment (william-gr) [inline src/freenas/usr/local/libexec/nas/generate_smb4_conf.py]: Dont you need to set NT1 in a `else`?

File: gui/services/migrations/0020_add_enable_smb1.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+
+def move_sysctl_min_protocol(apps, schemaeditor):
+    tunables = apps.get_model('system', 'tunable')
+    smb1_sysctl = tunables.filter(tun_type='SYSCTL',
+                                  tun_var='freenas.services.smb.config.server_min_protocol',
+                                  tun_value='NT1')
+
+    if smb1_sysctl.exists():
+        cifs.cifs_srv_enable_smb1 = True
```
Comment (william-gr) [inline gui/services/migrations/0020_add_enable_smb1.py]: You also  need to get cifs

apps.get_model('services', 'cifs').objects.order_by('-id')[0]

File: src/freenas/usr/local/libexec/nas/generate_smb4_conf.py (hunk 2 above)
Comment (anodos325) [reply src/freenas/usr/local/libexec/nas/generate_smb4_conf.py]: Samba defaults to allow NT1, but I agree it's better to explicitly set it. Will make that change.

**Comment by william-gr** [comment]
backport

**Comment by themylogin** [comment]
backport

**Comment by themylogin** [comment]
backport

**Comment by themylogin** [comment]
backport

**Comment by themylogin** [comment]
backport

**Pull request merged**